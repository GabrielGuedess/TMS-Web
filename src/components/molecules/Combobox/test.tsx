import { render } from '@testing-library/react';

import { ComboBox } from '.';

const setValue = jest.fn();
const setSearch = jest.fn();

describe('<ComboBox />', () => {
  it('should render the ComboBox correctly', () => {
    const { container } = render(
      <ComboBox
        label="País"
        search="Brazil"
        isLoading={false}
        setValue={setValue}
        setSearch={setSearch}
        placeholder="Selecione o País"
        emptyMessage="País não encontrado"
        placeholderCommand="Pesquise o País..."
        value={{ id: '1', description: 'Brazil' }}
        values={[{ id: '1', description: 'Brazil' }]}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
