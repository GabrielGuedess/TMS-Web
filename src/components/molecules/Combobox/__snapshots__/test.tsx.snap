// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComboBox /> should render the ComboBox correctly 1`] = `
<button
  aria-controls="radix-:r0:"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="group relative flex items-center justify-between rounded-lg border-2 border-zumthor-100 bg-white-lilac-50 p-3 text-sm font-medium text-comet-500 transition-all hover:bg-porcelain-50 focus:outline-none dark:border-shark-950 dark:bg-shark-950 dark:text-dark-300 dark:hover:opacity-60 focus:border-primary-300 dark:focus:border-primary-400 dark:data-[state="open"]:border-primary-400 data-[state="open"]:border-primary-300 focus:placeholder:opacity-100"
  data-state="closed"
  type="button"
>
  Brazil
  <label
    class="pointer-events-none absolute left-4 top-[-47%] z-10 translate-y-1/2 px-[3px] text-sm text-comet-400 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all group-focus:text-primary-300 group-data-[state="open"]:text-primary-400 dark:text-dark-300 before:dark:bg-smoke-950"
  >
    Pa√≠s
  </label>
</button>
`;
